src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs

```diff
@@ -695,11 +701,23 @@ private void SynchronizeContentTemplatedParent()
 			{
 				if (_contentTemplateRoot is IFrameworkElement binder)
 				{
-					var templatedParent = _contentTemplateRoot is ImplicitTextBlock
-						? this // ImplicitTextBlock is a special case that requires its TemplatedParent to be the ContentPresenter
-						: (this.TemplatedParent as IFrameworkElement)?.TemplatedParent;
+					binder.TemplatedParent = FindTemplatedParent();
 
-					binder.TemplatedParent = templatedParent;
+					object FindTemplatedParent()
+					{
+						// ImplicitTextBlock is a special case that requires its TemplatedParent to be the ContentPresenter
+						if (_contentTemplateRoot is ImplicitTextBlock) return this;
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6048912)